From 0139d8799964d1f925241d07d5ce26b041a99ce7 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@protonmail.com>
Date: Fri, 14 Feb 2025 16:06:58 -0700
Subject: [PATCH 1/1] Add 'executable' USE flag

Add 'executable' USE flag to toggle building the 'hnix' executable
(enabled by default)

Signed-off-by: hololeap <hololeap@protonmail.com>
--- a/hnix.cabal
+++ b/hnix.cabal
@@ -326,6 +326,10 @@ flag profiling
   manual: True
   default: False
 
+flag executable
+  description: Build the hnix executable
+  default: True
+
 common shared
   default-language: Haskell2010
   default-extensions:
@@ -499,6 +503,8 @@ library
 
 executable hnix
   import: shared
+  if !flag(executable)
+    buildable: False
   hs-source-dirs:
     main
   main-is: Main.hs
-- 
2.45.3

